src/core/or/circuitpadding.c

```diff
@@ -161,6 +161,113 @@ circpad_circuit_machineinfo_free_idx(circuit_t *circ, int idx)
   }
 }
 
+/**
+ * Return true if circpad has decided to hold the circuit open for additional
+ * padding. This function is used to take and retain ownership of certain
+ * types of circuits that have padding machines on them, that have been passed
+ * to circuit_mark_for_close().
+ *
+ * circuit_mark_for_close() calls this function to ask circpad if any padding
+ * machines want to keep the circuit open longer to pad.
+ *
+ * Any non-measurement circuit that was closed for a normal, non-error reason
+ * code may be held open for up to CIRCPAD_DELAY_INFINITE microseconds between
+ * network-driven cell events.
+ *
+ * After CIRCPAD_DELAY_INFINITE microseconds of silence on a circuit, this
+ * function will no longer hold it open (it will return 0 regardless of
+ * what the machines ask for, and thus circuit_expire_old_circuits_clientside()
+ * will close the circuit after roughly 1.25hr of idle time, maximum,
+ * regardless of the padding machine state.
+ */
+int
+circpad_is_using_circuit_for_padding(circuit_t *circ, int reason)
```

Comment:
It's still wrong for a function with a name like this to make changes on a circuit.  It should really just return 1 or 0, and let the caller invoke another function that adjusts the dirty timestamp and the purpose.

Reply:
(If it's going to have the current functionality, it really needs a different name.)

Reply:
How about circpad_marked_circuit_for_padding()? d9c864b7294273a5dd41e25fae1bd35571ebf52e